File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        int diff;
        for (int i = 0; i < nums.size(); i++) {
            diff = target - nums[i];
            if (hash.count(diff)) return {hash[diff], i};
            hash[nums[i]] = i;
        }
        return {0};
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1

        if nums[l] <= nums[r]:
            return nums[l]

        while nums[l] > nums[r]:
            r = r - 1
        
        return nums[r + 1]